.github/workflows/ci-lambda.yml

```diff
@@ -0,0 +1,87 @@
+name: CI - Python
+
+on:
+  push:
+    branches:
+      - master
+      - BFD-3524
+    paths:
+      - '**/*.py'
+      - '**/*requirement*.txt'
+  pull_request:
+    branches:
+      - master
+      - BFD-3524
+    paths:
+      - '**/*.py'
+      - '**/*requirement*.txt'
+  workflow_dispatch:
+
+jobs:
+  lambda-test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        id: code-checkout
+        uses: actions/checkout@v4
+        with:
+            fetch-depth: 2
+
+      - name: Set up Python
+        id: python-setup
+        uses: actions/setup-python@v5.1.1
+        with:
+          python-version: '3.11' # Specify the required Python version
```

Comment:
Another reason why we should try to standardize on a single version of python 😬

Reply:
Really, the reason why we hadn't standardized is because AWS was lagging behind supporting the latest versions of Python for their default runtime. I think we were stuck with 3.9, which would have been _fine_, but 3.12 was available and it had some noticeable improvements to type hinting syntax that would have been otherwise incompatible with 3.9 (as in, it straight up wouldn't have compiled).

We definitely should standardize on a version for our Lambdas running on the default runtime, but I personally think there's less of a need with Docker-based Lambdas. _Especially_ if we standardize on modern tools like `uv` and `ruff` (i.e. `uv` can generate virtualenvs based on a given Python version, `ruff` is an all-in-one linter and formatter that supports an array of `3.x` Python versions).